src/ekore/harmonics/__init__.py

```diff
@@ -2,266 +2,131 @@
-
-    """
-    sm31 = Sm31(n, sx[0], smx[0], smx[1], is_singlet)
-    return np.array(
-        [
-            S31(n, sx[0], sx[1], sx[2], sx[3]),
-            S211(n, sx[0], sx[1], sx[2]),
-            Sm22(n, sx[0], sx[1], smx[1], sm31, is_singlet),
-            Sm211(n, sx[0], sx[1], smx[0], is_singlet),
-            sm31,
-        ]
-    )
-
-
-@nb.njit(cache=True)
-def compute_cache(n, max_weight, is_singlet):
-    r"""Get the harmonics sums cache.
-
-    Parameters
-    ----------
-    n : complex
-        Mellin moment
-    max_weight : int
-        maximum weight to compute [2,3,4,5]
-    is_singlet : bool
-        symmetry factor: True for singlet like quantities (:math:`\eta=(-1)^N = 1`),
-        False for non-singlet like quantities (:math:`\eta=(-1)^N=-1`)
-
-    Returns
-    -------
-    list
-        harmonic sums cache. At |N3LO| it contains:
-
-        .. math ::
-            [[S_1,S_{-1}],
-            [S_2,S_{-2}],
-            [S_{3}, S_{2,1}, S_{2,-1}, S_{-2,1}, S_{-2,-1}, S_{-3}],
-            [S_{4}, S_{3,1}, S_{2,1,1}, S_{-2,-2}, S_{-3, 1}, S_{-4}],]
-            [S_{5}, S_{-5}]
-
-    """
-    # max number of harmonics sum of a given weight for a given max weight.
-    n_max_sums_weight = {2: 1, 3: 3, 4: 7, 5: 7}
-    sx = base_harmonics_cache(n, is_singlet, max_weight, n_max_sums_weight[max_weight])
-    if max_weight == 3:
-        # Add Sm21 to cache
-        sx[2, 1] = Sm21(n, sx[0, 0], sx[0, -1], is_singlet)
-    elif max_weight >= 4:
-        # Add weight 3 and 4 to cache
-        sx[2, 1:-2] = s3x(n, sx[:, 0], sx[:, -1], is_singlet)
-        sx[3, 1:-1] = s4x(n, sx[:, 0], sx[:, -1], is_singlet)
-    # return list of list keeping the non zero values
-    return [[el for el in sx_list if el != 0] for sx_list in sx]
-
-
-@nb.njit(cache=True)
-def compute_qed_ns_cache(n, s1):
-    r"""Get the harmonics sums cache needed for the qed non singlet AD.
-
-    Parameters
-    ----------
-    n : complex
-        Mellin moment
-    s1 : float
-        harmonic sum :math:`S_1(N)`
-
-    Returns
-    -------
-    list
-        harmonic sums cache. It contains:
-
-        .. math ::
-            [S_1(n/2),
-            S_2(n/2),
-            S_{3}(n/2),
-            S_1((n+1)/2),
-            S_2((n+1)/2),
-            S_{3}((n+1)/2),
-            g_3(n),
-            g_3(n+2)]
-
-    """
-    s1h = S1(n / 2.0)
-    sx_qed_ns = [s1h]
-    S2h = S2(n / 2.0)
-    sx_qed_ns.append(S2h)
-    S3h = S3(n / 2.0)
-    sx_qed_ns.append(S3h)
-    S1p1h = S1((n + 1.0) / 2.0)
-    sx_qed_ns.append(S1p1h)
-    S2p1h = S2((n + 1.0) / 2.0)
-    sx_qed_ns.append(S2p1h)
-    S3p1h = S3((n + 1.0) / 2.0)
-    sx_qed_ns.append(S3p1h)
-    g3N = g_functions.mellin_g3(n, s1)
-    sx_qed_ns.append(g3N)
-    S1p2 = polygamma.recursive_harmonic_sum(s1, n, 2, 1)
-    g3Np2 = g_functions.mellin_g3(n + 2.0, S1p2)
-    sx_qed_ns.append(g3Np2)
-    return sx_qed_ns
+# TODO: CLEAN
```

Comment:
Remember doing this before attempting the merge.

Reply:
Sure I marked them explicitly!

Reply:
I know. I just opened the conversation to make it explicit even in the GitHub interface :)